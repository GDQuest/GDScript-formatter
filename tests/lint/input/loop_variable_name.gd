var array = [1, 2, 3, 4, 5]

#region Good

func good():
	for i in array:
		print(i)

	for some_value in array:
		print(some_value)

	for _private in array:
		print(_private)

	for _some_value in array:
		print(_some_value)

#endregion

#region Bad

func bad():
	for A in array:
		print(A)

	for Something in array:
		print(Something)

	for _B in array:
		print(_B)

	for someValue in array:
		print(someValue)

#endregion
