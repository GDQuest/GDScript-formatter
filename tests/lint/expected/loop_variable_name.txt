tests/lint/input/loop_variable_name.gd:23:loop-variable-name:error: Loop variable 'A' should be in snake_case format
tests/lint/input/loop_variable_name.gd:26:loop-variable-name:error: Loop variable 'Something' should be in snake_case format
tests/lint/input/loop_variable_name.gd:29:loop-variable-name:error: Loop variable '_B' should be in snake_case format
tests/lint/input/loop_variable_name.gd:32:loop-variable-name:error: Loop variable 'someValue' should be in snake_case format